app/models/goal.py

```diff
@@ -3,3 +3,12 @@
 
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal")
+
+    # helper function to structure request response json
+    def make_goal_dict(self):
```

Comment:
Great helper method.  You could also create a class-method as a helper which takes a dictionary as an argument and creates an instance of `Goal` from that dict.

```py
    @classmethod
    def make_from_dict(cls, dict):
        return cls(title = dict["title"])
```